HPCC-20530 Log basic information (size, last modification date) if spray source file(s).

AttilaVamos opened the pull request:
Add code to log basic input file parameters (name, size, modification date)
for spray/copy/replication operations

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->
Testing maually with ECLWatch, DFUPlus with spraying simple and multiple files.
<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 198a4ce1c575120492102b881b1f1c15e6e12e3b
Build: success
Install hpccsystems-platform-community_7.0.0-rc2.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |806 |806 |0 |
|test (thor) |730 |730 |0 |
|test (roxie) |879 |879 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  188 sec (00:03:08) |   95 sec (00:01:35) |   22 sec (00:00:22) |   44 sec (00:00:44) | 1308 sec (00:21:48) |   18 sec (00:00:18) | 1684 sec (00:28:04) |

AttilaVamos commented:
@jakesmith please review.

HPCCSmoketest commented:
This PR (11700) is in the 1/1 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
This PR (11700) is in the 1/1 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
This PR (11700) is in the 1/2 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
This PR (11700) is in the 1/2 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
This PR (11700) is in the 1/2 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
This PR (11700) is in the 1/2 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
This PR (11700) is in the 1/2 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
This PR (11700) is in the 1/2 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
This PR (11700) is in the 1/2 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
This PR (11700) is in the 1/2 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
This PR (11700) is in the 1/2 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
This PR (11700) is in the 1/1 position of the queue. Estimated start time is after ~3.0 hour(s)

HPCCSmoketest commented:
2/2. Process PR-11700, label: uHPCC-20530-improvement-7.0.0 starts now.

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: Ubuntu 16.04 (Linux 4.4.0-135-generic)
Sha: 198a4ce1c575120492102b881b1f1c15e6e12e3b
Build: success
Install hpccsystems-platform-community_7.0.0-rc2xenial_amd64.deb
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |806 |806 |0 |
|test (thor) |730 |726 |4 |
|test (roxie) |879 |879 |0 |


[Errors:](https://gist.github.com/d11e9bf0c0a98a37ba2ed2e43a1775b7)
- test
  - thor
    - 243\. [genjoin3](https://gist.github.com/HPCCSmoketest/d11e9bf0c0a98a37ba2ed2e43a1775b7#file-genjoin3-w20180915-215244-log), 357\. [keydiff1](https://gist.github.com/HPCCSmoketest/d11e9bf0c0a98a37ba2ed2e43a1775b7#file-keydiff1-w20180915-220445-log), 399\. [loopagain2](https://gist.github.com/HPCCSmoketest/d11e9bf0c0a98a37ba2ed2e43a1775b7#file-loopagain2-w20180915-221738-log), 668\. [teststdlibrary](https://gist.github.com/HPCCSmoketest/d11e9bf0c0a98a37ba2ed2e43a1775b7#file-teststdlibrary-w20180915-223226-log), 

HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   51 sec (00:00:51) | 1703 sec (00:28:23) |  264 sec (00:04:24) |  223 sec (00:03:43) |  266 sec (00:04:26) | 9120 sec (02:32:00) |   32 sec (00:00:32) |11659 sec (03:14:19) |

HPCCSmoketest commented:
This PR (11700) is in the 1/1 position of the queue. Estimated start time is after ~3.0 hour(s)

HPCCSmoketest commented:
2/2. Process of PR-11700, label: uHPCC-20530-improvement-7.0.0 starts now. Estimated completion time is ~ 3.0 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: Ubuntu 16.04 (Linux 4.4.0-135-generic)
Sha: 198a4ce1c575120492102b881b1f1c15e6e12e3b
Build: success
Install hpccsystems-platform-community_7.0.0-rc2xenial_amd64.deb
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |806 |806 |0 |
|test (thor) |730 |726 |4 |
|test (roxie) |879 |879 |0 |


[Errors:](https://gist.github.com/6c1d95846f1d6631718df52ab74b271f)
- test
  - thor
    - 243\. [genjoin3](https://gist.github.com/HPCCSmoketest/6c1d95846f1d6631718df52ab74b271f#file-genjoin3-w20180916-054655-log), 357\. [keydiff1](https://gist.github.com/HPCCSmoketest/6c1d95846f1d6631718df52ab74b271f#file-keydiff1-w20180916-055921-log), 399\. [loopagain2](https://gist.github.com/HPCCSmoketest/6c1d95846f1d6631718df52ab74b271f#file-loopagain2-w20180916-061309-log), 668\. [teststdlibrary](https://gist.github.com/HPCCSmoketest/6c1d95846f1d6631718df52ab74b271f#file-teststdlibrary-w20180916-062826-2-log), 

HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   55 sec (00:00:55) | 1843 sec (00:30:43) |  262 sec (00:04:22) |  167 sec (00:02:47) |  211 sec (00:03:31) | 9522 sec (02:38:42) |   33 sec (00:00:33) |12093 sec (03:21:33) |

HPCCSmoketest commented:
This PR (11700) is in the 1/1 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
2/2. Process of PR-11700, label: uHPCC-20530-improvement-7.0.0 starts now. Estimated completion time is ~ 0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 198a4ce1c575120492102b881b1f1c15e6e12e3b
Build: success
Install hpccsystems-platform-community_7.0.0-rc2.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |806 |806 |0 |
|test (thor) |730 |730 |0 |
|test (roxie) |879 |879 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   17 sec (00:00:17) |  191 sec (00:03:11) |   94 sec (00:01:34) |   21 sec (00:00:21) |   43 sec (00:00:43) | 1528 sec (00:25:28) |   18 sec (00:00:18) | 1912 sec (00:31:52) |

HPCCSmoketest commented:
This PR (11700) is in the 1/1 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
2/2. Process of PR-11700, label: HPCC-20530-improvement-7.0.0 starts now. Estimated completion time is ~ 0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 198a4ce1c575120492102b881b1f1c15e6e12e3b
Build: success
Install hpccsystems-platform-community_7.0.0-rc2.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |806 |806 |0 |
|test (thor) |730 |730 |0 |
|test (roxie) |879 |879 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   35 sec (00:00:35) |  187 sec (00:03:07) |   95 sec (00:01:35) |   19 sec (00:00:19) |   41 sec (00:00:41) | 1700 sec (00:28:20) |   18 sec (00:00:18) | 2095 sec (00:34:55) |

jakesmith commented on dali/ft/filecopy.cpp:
```diff
@@ -543,6 +543,8 @@ int FileSizeThread::run()
                 }
                 cur->size = thisSize;
+                CDateTime create, update;
+                thisFile->getTime(&create, &cur->modifiedTime, &update);
```
you can pass nullptr for unwanted create and update fields.

jakesmith commented on dali/ft/filecopy.cpp:
```diff
@@ -1793,12 +1795,18 @@ void FileSprayer::gatherFileSizes(bool errorIfMissing)
+            LOG(MCdebugProgress, job, "%9d:%s (size: %llu bytes, last modified: %s %s)",
+                                       idx2, cur.filename.getTail(tailStr.clear()).str(), cur.size,
+                                       cur.modifiedTime.getDateString(dateStr.clear(), true).str(),
+                                       cur.modifiedTime.getTimeString(timeStr.clear(), true).str()
```
CDateTime::getString() would do in 1 call and is slightly more efficient.

AttilaVamos replied:
Changed

jakesmith commented on dali/ft/filecopy.cpp:
```diff
@@ -1793,12 +1795,18 @@ void FileSprayer::gatherFileSizes(bool errorIfMissing)
         {
             FilePartInfo & cur = sources.item(idx2);
+
+            LOG(MCdebugProgress, job, "%9d:%s (size: %llu bytes, last modified: %s %s)",
```
picky: %9u more correct as idx2 is unsigned

AttilaVamos replied:
Changed

jakesmith commented on dali/ft/filecopy.cpp:
```diff
@@ -1810,6 +1818,8 @@ void FileSprayer::afterGatherFileSizes()
             }
         }
+        LOG(MCdebugProgress, job, "----------------------------------------------");
+        LOG(MCdebugProgress, job, "All together: %llu bytes in %d file(s)", totalSize, sources.ordinality());
```
picky: %u instead of %d

AttilaVamos replied:
Changed

jakesmith commented on dali/ft/filecopy.cpp:
```diff
@@ -2563,7 +2573,8 @@ void FileSprayer::setSource(IFileDescriptor * source, unsigned copy, unsigned mi
                 next.extractExtra(*part);
-                next.size = multi.getSize(i);
+                if (copy)
+                    next.size = multi.getSize(i);
```
Not sure I understand this change.. is it related to this JIRA?

AttilaVamos replied:
Yes, it is related and I will explain.

AttilaVamos replied:
So, the reason for that check was: don't set the file part size here - force to check disk.
Later I realised there were a couple, redundant gatherFileSizes(false); function calls in the setSource() methods (like this). I removed them, based on the spray() always call that function with errorIfMissing = true.

jakesmith commented:
@AttilaVamos - added some comments/questions.

This change will mean that it logs modification dates of the sources before it starts spraying, would that have been enough to diagnose the problem of the file changing during the spray ?

I think as it stands, you'd have to look at the physical files to diagnose and see that the modification date was ahead of the logged modification date, but you would not be able to tell when it changed, e.g. might have changed after spray completed.. 

I think you might need to log (or spot) the modification data/size at the end of the spray too to help diagnose source files changing during spray..

HPCCSmoketest commented:
This PR (11700) is in the 1/1 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
2/2. Process of PR-11700, label: HPCC-20530-improvement-7.0.0 starts now. Estimated completion time is ~ 0.5 hour(s)

HPCCSmoketest commented:
1/1. Process of PR-11700, label: HPCC-20530-improvement-7.0.0 starts now. Estimated completion time is ~ 0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 642f91287cdddd0d2d23918fbfad6603e6229f83
Build: success
Install hpccsystems-platform-community_7.0.0-rc2.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |806 |806 |0 |
|test (thor) |730 |730 |0 |
|test (roxie) |879 |879 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   22 sec (00:00:22) |  188 sec (00:03:08) |   94 sec (00:01:34) |   20 sec (00:00:20) |   42 sec (00:00:42) | 1522 sec (00:25:22) |   20 sec (00:00:20) | 1908 sec (00:31:48) |

jakesmith commented on dali/ft/filecopy.cpp:
```diff
@@ -2520,18 +2528,12 @@ void FileSprayer::setSource(IDistributedFile * source)
     setSource(source, 0, 1);
-
-    //Now get the size of the files directly (to check they exist).  If they don't exist then switch to the backup instead.
-    gatherFileSizes(false);
```
This change is unrelated really and I think removing it may cause problems/changed the behaviour..
This passes false to gatherFileSizes and the method looks at copies and does switches to one of them (that's what the comment is about)..
What happens if primary not found here and mirror copy(ies) used...

I think the gatherFileSize(true) in spray() is a NOP, if this gatherFileSizes(false) call succeeded in getting all sizes, so not a waste to recall.

There may be an issue but it should be tackled in a separate JIRA..
I think what should happen is that it's called once, but that it's reworked to look at all copies and error only if can't find any..

AttilaVamos replied:
Point taken, restored.

jakesmith commented on dali/ft/filecopy.cpp:
```diff
@@ -2917,6 +2920,7 @@ void FileSprayer::spray()
 
 
+    //Get the size of the source files directly (to check they exist).
     gatherFileSizes(true);
```
without having called getFileSizes(false) earlier, this will now fail if primary copies are missing but replicates exist.
See other comment where getFileSizes(false) call removed.

AttilaVamos replied:
Reverted

jakesmith commented:
@AttilaVamos - have some concerns about removing the gahterFileSizes(false) calls - see comments, plus unrelated to this JIRA, so if there's an issue here, should be in a separate JIRA/PR.

jakesmith commented:
> .. We agreed to open another JIRA to make source file check at the end of spray and fail it if any source file changed during the operation.

@AttilaVamos - can you link that new JIRA to this JIRA ?

HPCCSmoketest commented:
1/3. Process of PR-11700, label: HPCC-20530-improvement-7.0.0 starts now. Estimated completion time is ~ 0.5 hour(s)

AttilaVamos commented:
@jakesmith comments are addressed, so function calls restored.
The new JIRA for fail spray if files changed is https://track.hpccsystems.com/browse/HPCC-20659

jakesmith commented:
@AttilaVamos looks good, please squash.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 7d454f0addb8fc603544bf80bc2b249b27d6b6de
Build: success
Install hpccsystems-platform-community_7.0.0-rc2.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |806 |806 |0 |
|test (thor) |730 |730 |0 |
|test (roxie) |879 |879 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   35 sec (00:00:35) |  188 sec (00:03:08) |   97 sec (00:01:37) |   20 sec (00:00:20) |   42 sec (00:00:42) | 1463 sec (00:24:23) |   20 sec (00:00:20) | 1865 sec (00:31:05) |

HPCCSmoketest commented:
1/2. Process of PR-11700, label: HPCC-20530-improvement-7.0.0 starts now. Estimated completion time is ~ 0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: cd4eb58b4184226bcb7a9966eaf964f383582d01
Build: success
Install hpccsystems-platform-community_7.0.0-rc2.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |806 |806 |0 |
|test (thor) |730 |730 |0 |
|test (roxie) |879 |879 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   14 sec (00:00:14) |  185 sec (00:03:05) |   95 sec (00:01:35) |   19 sec (00:00:19) |   42 sec (00:00:42) | 1492 sec (00:24:52) |   20 sec (00:00:20) | 1867 sec (00:31:07) |

AttilaVamos committed:
HPCC-20530 Log basic information (size, last modification date) if spray source file(s).

Add code to log basic input file parameters (name, size, modification date)
for spray/copy/replication operations

Testing maually with ECLWatch, DFUPlus with spraying simple and multiple files.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

AttilaVamos commented:
@jakesmith done

HPCCSmoketest commented:
1/2. Process of PR-11700, label: HPCC-20530-improvement-7.0.0 starts now. Estimated completion time is ~ 0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: cd8d1c8d378cdda83562189f5d06e7b05a074d7f
Build: success
Install hpccsystems-platform-community_7.0.0-rc2.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |806 |806 |0 |
|test (thor) |730 |730 |0 |
|test (roxie) |879 |879 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   17 sec (00:00:17) |  188 sec (00:03:08) |   94 sec (00:01:34) |   22 sec (00:00:22) |   44 sec (00:00:44) | 1467 sec (00:24:27) |   19 sec (00:00:19) | 1851 sec (00:30:51) |
